File: routes/parsedModel.js
const express = require("express");
const modelRouter = express.Router();
const ParsedModel = require('../model');
modelRouter.get('/parsedModels', async (req, res) => {
    const page = parseInt(req.query.page) || 1;
    const limit = parseInt(req.query.limit) || 10;

    const startIndex = (page - 1) * limit;
    const endIndex = page * limit;

    try {
        const parsedModels = await ParsedModel.find().skip(startIndex).limit(limit);
        res.status(200).json({
            parsedModels
        });
    } catch (error) {
        res.status(500).json({
            error: 'Internal server error'
        });
    }
});
modelRouter.get('/parsing', async (req, res) => {
    
});




// Create a new parsedModel
modelRouter.post('/parsedModel-create', async (req, res) => {
    
    const parsedModel = new ParsedModel({
        id: req.body.id,
        linkId: req.body.linkId,
        title: req.body.title,
        price: req.body.price,
        color: req.body.color,
        memory: req.body.memory,
        images: req.body.images,
        specification: req.body.specification,
        reviews: req.body.reviews
    });

    try {
        const savedParsedModel = await parsedModel.save();
        res.status(201).json({
            parsedModel: savedParsedModel
        });
    } catch (error) {
        res.status(400).json({
            error: 'Invalid request data'
        });
    }
});

// Get all parsedModels
modelRouter.get('/parsedModels', async (req, res) => {
    try {
        const parsedModels = await ParsedModel.find();
        res.status(200).json({
            parsedModels
        });
    } catch (error) {
        res.status(500).json({
            error: 'Internal server error'
        });
    }
});

// // Get a single parsedModel by id
// modelRouter.get('/parsedModel/:id', async (req, res) => {
//     try {
//         const parsedModel = await ParsedModel.findById(req.params.id);
//         res.status(200).json({
//             parsedModel
//         });
//     } catch (error) {
//         res.status(404).json({
//             error: 'ParsedModel not found'
//         });
//     }
// });

// Update a parsedModel
modelRouter.put('/parsedModel/:id', async (req, res) => {
    try {
        
        console.log(req.body);
        const updatedParsedModel = await ParsedModel.updateOne(
            {id: req.params.id}, {
                title: req.body.title,
                price: req.body.price,
            }
        );
        res.status(200).json({
            parsedModel: updatedParsedModel
        });
    } catch (error) {
        console.log(error);
        res.status(404).json({
            error: 'ParsedModel not found'
        });
    }
});





modelRouter.delete('/parsedmodel-delete/:id', async (req, res) => {
    const id = req.params.id;
    try {
        const result = await ParsedModel.deleteOne({ id: id });
        if (result.deletedCount === 0) {
            res.status(404).json({
                error: 'not found'
            });
        }
        res.status(200).json({
            successful: 'ParsedModel deleted'
        });
    } catch (error) {
        res.status(500).json({
            error: 'Server error'
        });
    }
});

module.exports = modelRouter

File: flutter_application_technodom/lib/parsed_model.dart
import 'dart:convert';

import 'review.dart';
import 'specification.dart';

class ParsedModel {
  final String id;
  final String linkId;
  String title;
  String price;
  final List<String> color;
  final List<String> memory;
  final List<String> images;
  final List<Specification> specification;
  final List<Review> reviews;
  ParsedModel({
    required this.id,
    required this.linkId,
    required this.title,
    required this.price,
    required this.color,
    required this.memory,
    required this.images,
    required this.specification,
    required this.reviews,
  });

  Map<String, dynamic> toMap() {
    final result = <String, dynamic>{};
  
    result.addAll({'id': id});
    result.addAll({'linkId': linkId});
    result.addAll({'title': title});
    result.addAll({'price': price});
    result.addAll({'color': color});
    result.addAll({'memory': memory});
    result.addAll({'images': images});
    result.addAll({'specification': specification.map((x) => x.toMap()).toList()});
    result.addAll({'reviews': reviews.map((x) => x.toMap()).toList()});
  
    return result;
  }

  factory ParsedModel.fromMap(Map<String, dynamic> map) {
    return ParsedModel(
      id: map['id'] ?? '',
      linkId: map['linkId'] ?? '',
      title: map['title'] ?? '',
      price: map['price'] ?? '',
      color: List<String>.from(map['color']),
      memory: List<String>.from(map['memory']),
      images: List<String>.from(map['images']),
      specification: List<Specification>.from(map['specification']?.map((x) => Specification.fromMap(x))),
      reviews: List<Review>.from(map['reviews']?.map((x) => Review.fromMap(x))),
    );
  }

  String toJson() => json.encode(toMap());

  factory ParsedModel.fromJson(String source) => ParsedModel.fromMap(json.decode(source));
}
